src/util/std_code.h

```diff
@@ -1378,15 +1379,15 @@ inline bool can_cast_expr<side_effect_expr_nondett>(const exprt &base)
 inline side_effect_expr_nondett &to_side_effect_expr_nondet(exprt &expr)
 {
   auto &side_effect_expr_nondet=to_side_effect_expr(expr);
-  assert(side_effect_expr_nondet.get_statement()==ID_nondet);
+  CHECK_RETURN(side_effect_expr_nondet.get_statement()==ID_nondet);
```

Comment:
Ugh, that's one of the nastier cases... should arguably be `PRECONDITION`, but it happens after a conversion... I'd just put it on the line above and do a `PRECONDITION(expr.get(ID_statement) == ID_nondet)`

While you're at it, I'd also add the `PRECONDITION(expr.id() == ID_side_effect)` as in the examples below